Author: Philipp Hahn <pmhahn@debian.org>

Fix shell quoting.
--- a/contrib/action.script
+++ b/contrib/action.script
@@ -4,43 +4,43 @@
 
 scan_dir=/var/spool/scan
 datetime=`date +%F_%H%M%S`
-filename=scan-$datetime
+filename="scan-$datetime"
 send_to=email@address.com
 
-case $SCANBD_ACTION in
+case "$SCANBD_ACTION" in
  pdf)
   logger -t "scanbd: $0" "$SCANBD_ACTION - scanning and converting to ps"
-  scanimage -d $SCANBD_DEVICE --mode Color --resolution 150 --depth 16 |
-convert -density 150 - $scan_dir/$filename.ps
+  scanimage -d "$SCANBD_DEVICE" --mode Color --resolution 150 --depth 16 |
+   convert -density 150 - "$scan_dir/$filename.ps"
   logger -t "scanbd: $0" "$SCANBD_ACTION - converting ps to pdf"
-  ps2pdf  -sPAPERSIZE=a4 $scan_dir/$filename.ps $scan_dir/$filename.pdf
-  rm $scan_dir/$filename.ps
+  ps2pdf  -sPAPERSIZE=a4 "$scan_dir/$filename.ps" "$scan_dir/$filename.pdf"
+  rm "$scan_dir/$filename.ps"
   ;;
  scan)
-  scanimage -d $SCANBD_DEVICE --resolution 150 --mode Color --depth 16
---format=tiff > $scan_dir/$filename.tiff
+  scanimage -d "$SCANBD_DEVICE" --resolution 150 --mode Color --depth 16 --format=tiff > "$scan_dir/$filename.tiff"
   ;;
  copy)
   logger -t "scanbd: $0" "$SCANBD_ACTION - scanning"
-  scanimage -d $SCANBD_DEVICE --resolution 300 --depth 16--format=tiff > $scan_dir/$filename.tiff
+  scanimage -d "$SCANBD_DEVICE" --resolution 300 --depth 16--format=tiff > "$scan_dir/$filename.tiff"
 # | convert -density 300 - $scan_dir/$filename.ps
   logger -t "scanbd: $0" "$SCANBD_ACTION - printing"
-  lp $scan_dir/$filename.tiff
-  rm $scan_dir/$filename.ps
+  lp "$scan_dir/$filename.tiff"
+  rm "$scan_dir/$filename.tiff"
   ;;
 email)
   logger -t "scanbd: $0" "$SCANBD_ACTION - scanning and converting to ps"
-  scanimage -d $SCANBD_DEVICE --mode Color --resolution 150 --depth 16|
-convert -density 150 - $scan_dir/$filename.ps
+  scanimage -d "$SCANBD_DEVICE" --mode Color --resolution 150 --depth 16 |
+   convert -density 150 - "$scan_dir/$filename.ps"
   logger -t "scanbd: $0" "$SCANBD_ACTION - converting ps to pdf"
-  ps2pdf  -sPAPERSIZE=a4 $scan_dir/$filename.ps $scan_dir/$filename.pdf
+  ps2pdf  -sPAPERSIZE=a4 "$scan_dir/$filename.ps" "$scan_dir/$filename.pdf"
   logger -t "scanbd: $0" "$SCANBD_ACTION - sending pdf"
   (cat <<EOCAT
 Document scanned on Canon LiDE 110
 EOCAT
-uuencode $scan_dir/$filename.pdf $scan_dir/$filename.pdf ) | mail -a
-"From: Canon scanner <some@email.sk>" -s "Document $filename" $send_to
-  rm $scan_dir/$filename.ps
+uuencode "$scan_dir/$filename.pdf" "$scan_dir/$filename.pdf" ) | mail -a
+"From: Canon scanner <some@email.sk>" -s "Document $filename" "$send_to"
+  rm "$scan_dir/$filename.ps"
+  rm "$scan_dir/$filename.pdf"
   ;;
 *)
   logger -t "scanbd: $0" "Unhandled action request."
--- a/contrib/actions.script
+++ b/contrib/actions.script
@@ -47,87 +47,87 @@ EMAIL="someone@example.org" # email addr
 USER="user" # useful when saving scans in NFS shared directory
 GROUP="group" # useful when saving scans in NFS shared directory
 
-export MAGICK_TMPDIR=${TMPDIR} # Override default cache of convert command. Needed for systems with very small /tmp (e.g. ramdisk).
+export MAGICK_TMPDIR="${TMPDIR}" # Override default cache of convert command. Needed for systems with very small /tmp (e.g. ramdisk).
 
-case $SCANBD_ACTION in
+case "$SCANBD_ACTION" in
 
     scan)
 	STARTANEW=`find ${TMPDIR} -mindepth 1 -maxdepth 1 -name "${PAGEBASE}*.tif" -mmin -${TIMEOUT} 2> /dev/null | wc -l`
 
 	if [ "0${STARTANEW}" -eq 0 ]; then
-		scanimage -d $SCANBD_DEVICE \
+		scanimage -d "$SCANBD_DEVICE" \
 		--icc-profile=/usr/share/color/icc/sRGB.icm --format=tiff \
 		--mode=Color --depth=8 --resolution=300 \
-		> ${TMPDIR}${FILENAME}.tif
+		> "${TMPDIR}${FILENAME}.tif"
 
-		convert -rotate 180 ${TMPDIR}${FILENAME}.tif ${SCANDIR}${FILENAME}.jpg
-		chown ${USER}:${GROUP} ${SCANDIR}${FILENAME}.jpg
+		convert -rotate 180 "${TMPDIR}${FILENAME}.tif" "${SCANDIR}${FILENAME}.jpg"
+		chown "${USER}:${GROUP}" "${SCANDIR}${FILENAME}.jpg"
 
-		rm ${TMPDIR}${FILENAME}.tif
+		rm "${TMPDIR}${FILENAME}.tif"
 	else
-		NEXTPAGE=`expr 1 + \`ls -1 ${TMPDIR}${PAGEBASE}*.tif | wc -l\``
+		NEXTPAGE=`expr 1 + \`ls -1 "${TMPDIR}${PAGEBASE}"*.tif | wc -l\``
 
-		scanimage -d $SCANBD_DEVICE \
+		scanimage -d "$SCANBD_DEVICE" \
 		--icc-profile=/usr/share/color/icc/sRGB.icm --format=tiff \
 		--mode=Color --depth=8 --resolution=150 \
-		> ${TMPDIR}${PAGEBASE}`printf %.5d ${NEXTPAGE}`.tif
+		> "${TMPDIR}${PAGEBASE}`printf %.5d ${NEXTPAGE}`.tif"
 	fi
     ;;
 
     copy)
-	scanimage -d $SCANBD_DEVICE --format=tiff --mode=Gray --depth=8 \
+	scanimage -d "$SCANBD_DEVICE" --format=tiff --mode=Gray --depth=8 \
 	--resolution=300 \
 	| convert - -density 300 ps:- \
-	| lp -d ${PRINTER}
+	| lp -d "${PRINTER}"
     ;;
 
     email)
-	STARTANEW=`find ${TMPDIR} -mindepth 1 -maxdepth 1 -name "${PAGEBASE}*.tif" -mmin -${TIMEOUT} 2> /dev/null | wc -l`
+	STARTANEW=`find "${TMPDIR}" -mindepth 1 -maxdepth 1 -name "${PAGEBASE}*.tif" -mmin -${TIMEOUT} 2> /dev/null | wc -l`
 
 	if [ "0${STARTANEW}" -eq 0 ]; then
-		scanimage -d $SCANBD_DEVICE \
+		scanimage -d "$SCANBD_DEVICE" \
 		--icc-profile=/usr/share/color/icc/sRGB.icm --format=tiff \
 		--mode=Color --depth=8 --resolution=150 \
-		> ${TMPDIR}${FILENAME}.tif
+		> "${TMPDIR}${FILENAME}.tif"
 
-		convert -rotate 180 ${TMPDIR}${FILENAME}.tif ${TMPDIR}${FILENAME}.jpg
-		touch ${TMPDIR}${FILENAME}.txt
-		mpack -s "Scanned document ${FILENAME}" -d ${TMPDIR}${FILENAME}.txt \
-		-o ${TMPDIR}${FILENAME}.eml ${TMPDIR}${FILENAME}.jpg 
+		convert -rotate 180 "${TMPDIR}${FILENAME}.tif" "${TMPDIR}${FILENAME}.jpg"
+		touch "${TMPDIR}${FILENAME}.txt"
+		mpack -s "Scanned document ${FILENAME}" -d "${TMPDIR}${FILENAME}.txt" \
+		-o "${TMPDIR}${FILENAME}.eml" "${TMPDIR}${FILENAME}.jpg"
 
-		cat ${TMPDIR}${FILENAME}.eml | sendmail -v ${EMAIL}
+		cat "${TMPDIR}${FILENAME}.eml" | sendmail -v "${EMAIL}"
 
-		rm ${TMPDIR}${FILENAME}.tif ${TMPDIR}${FILENAME}.eml ${TMPDIR}${FILENAME}.jpg ${TMPDIR}${FILENAME}.txt
+		rm "${TMPDIR}${FILENAME}.tif" "${TMPDIR}${FILENAME}.eml" "${TMPDIR}${FILENAME}.jpg" "${TMPDIR}${FILENAME}.txt"
 	else
 		convert -rotate 180 -units PixelsPerInch -density 150 -compress jpeg \
-		${TMPDIR}${PAGEBASE}*.tif ${TMPDIR}${FILENAME}.pdf
+		"${TMPDIR}${PAGEBASE}"*.tif "${TMPDIR}${FILENAME}.pdf"
 
-		touch ${TMPDIR}${FILENAME}.txt
-		mpack -s "Scanned document ${FILENAME}" -d ${TMPDIR}${FILENAME}.txt \
-		-o ${TMPDIR}${FILENAME}.eml ${TMPDIR}${FILENAME}.pdf
+		touch "${TMPDIR}${FILENAME}.txt"
+		mpack -s "Scanned document ${FILENAME}" -d "${TMPDIR}${FILENAME}.txt" \
+		-o "${TMPDIR}${FILENAME}.eml" "${TMPDIR}${FILENAME}.pdf"
 
-		cat ${TMPDIR}${FILENAME}.eml | sendmail -v ${EMAIL}
+		cat "${TMPDIR}${FILENAME}.eml" | sendmail -v "${EMAIL}"
 
-		rm ${TMPDIR}${PAGEBASE}*.tif ${TMPDIR}${FILENAME}.eml ${TMPDIR}${FILENAME}.pdf ${TMPDIR}${FILENAME}.txt
+		rm "${TMPDIR}${PAGEBASE}"*.tif "${TMPDIR}${FILENAME}.eml" "${TMPDIR}${FILENAME}.pdf" "${TMPDIR}${FILENAME}.txt"
 	fi
     ;;
 
     file)
-	STARTANEW=`find ${TMPDIR} -mindepth 1 -maxdepth 1 -name "${PAGEBASE}*.tif" -mmin -${TIMEOUT} 2> /dev/null | wc -l`
+	STARTANEW=`find "${TMPDIR}" -mindepth 1 -maxdepth 1 -name "${PAGEBASE}*.tif" -mmin -${TIMEOUT} 2> /dev/null | wc -l`
 
 	if [ "0${STARTANEW}" -eq 0 ]; then
-		rm ${TMPDIR}${PAGEBASE}*.tif &> /dev/null
+		rm "${TMPDIR}${PAGEBASE}"*.tif &> /dev/null
 
-		scanimage -d $SCANBD_DEVICE \
+		scanimage -d "$SCANBD_DEVICE" \
 		--icc-profile=/usr/share/color/icc/sRGB.icm --format=tiff \
 		--mode=Color --depth=8 --resolution=150 \
-		> ${TMPDIR}${PAGEBASE}00001.tif
+		> "${TMPDIR}${PAGEBASE}00001.tif"
 	else
 		convert -rotate 180 -units PixelsPerInch -density 150 -compress jpeg \
-		${TMPDIR}${PAGEBASE}*.tif ${SCANDIR}${FILENAME}.pdf
-		chown ${USER}:${GROUP} ${SCANDIR}${FILENAME}.pdf
+		"${TMPDIR}${PAGEBASE}"*.tif "${SCANDIR}${FILENAME}.pdf"
+		chown "${USER}:${GROUP}" "${SCANDIR}${FILENAME}.pdf"
 
-		rm ${TMPDIR}${PAGEBASE}*.tif
+		rm "${TMPDIR}${PAGEBASE}"*.tif
 	fi
     ;;
 
--- a/scripts/example.script
+++ b/scripts/example.script
@@ -23,6 +23,6 @@
 
 logger -t "scanbd: $0" "Begin of $SCANBD_ACTION for device $SCANBD_DEVICE"
 
-scanimage -d $SCANBD_DEVICE  2>&1 > /tmp/scanner.test
+scanimage -d "$SCANBD_DEVICE"  2>&1 > /tmp/scanner.test
 
-logger -t "scanbd: $0" "End   of $SCANBD_ACTION for device $SCANBD_DEVICE"
\ No newline at end of file
+logger -t "scanbd: $0" "End   of $SCANBD_ACTION for device $SCANBD_DEVICE"
--- a/scripts/scanadf.script
+++ b/scripts/scanadf.script
@@ -23,7 +23,7 @@
 
 logger -t "scanbd: $0" "Begin of $SCANBD_ACTION for device $SCANBD_DEVICE"
 
-scanner=$(echo $SCANBD_DEVICE | cut -d ':' -f 1)
-scanadf -d $SCANBD_DEVICE -o /tmp/${scanner}_${SCANBD_FUNCTION_MODE}_${SCANBD_FUNCTION} 2>&1 
+scanner=$(echo "$SCANBD_DEVICE" | cut -d ':' -f 1)
+scanadf -d "$SCANBD_DEVICE" -o "/tmp/${scanner}_${SCANBD_FUNCTION_MODE}_${SCANBD_FUNCTION}" 2>&1
 
 logger -t "scanbd: $0" "End   of $SCANBD_ACTION for device $SCANBD_DEVICE"
--- a/contrib/insert.script
+++ b/contrib/insert.script
@@ -18,7 +18,6 @@
 logger="/usr/bin/logger"
 
 #$logger -t "scanbd: $0" "SCANBD_DEVICE=$SCANBD_DEVICE SCANBD_ACTION=$SCANBD_ACTION"
-$logger -t "scanbd: $0" "Loading firmware for $SCANBD_DEVICE"
-/usr/bin/scanimage -L 2>&1 | $logger -t "scanbd: $0"
-$logger -t "scanbd: $0" "Done loading firmware for $SCANBD_DEVICE"
-
+"$logger" -t "scanbd: $0" "Loading firmware for $SCANBD_DEVICE"
+/usr/bin/scanimage -L 2>&1 | "$logger" -t "scanbd: $0"
+"$logger" -t "scanbd: $0" "Done loading firmware for $SCANBD_DEVICE"
