Author: Rolf Leggewie <foss@rolf.leggewie.biz>
Bug: https://sourceforge.net/p/scanbd/tickets/1/

Scripts should live under /usr not in /etc

Index: scanbd/src/scanbd/config.c
===================================================================
--- scanbd.orig/src/scanbd/config.c	2017-10-26 13:40:29.994416647 +0800
+++ scanbd/src/scanbd/config.c	2017-10-26 13:40:29.990416626 +0800
@@ -169,8 +169,8 @@
         const char* scriptdir =  cfg_getstr(cfg_sec_global, C_SCRIPTDIR);
 
         if(!scriptdir || (strlen(scriptdir) == 0)) {
-            // scriptdir is not set, script is relative to SCANBD_CFG_DIR
-            snprintf(script_abs, PATH_MAX, "%s/%s", SCANBD_CFG_DIR, script);
+            // scriptdir is not set, look for scripts in /usr/share/scanbd/scripts
+            snprintf(script_abs, PATH_MAX, "/usr/share/scanbd/scripts/%s", script);
         } else if (scriptdir[0] == '/') {
             // scriptdir is an absolute path
             snprintf(script_abs, PATH_MAX, "%s/%s", scriptdir, script);
