Author: Rolf Leggewie <foss@rolf.leggewie.biz>
Bug: https://sourceforge.net/p/scanbd/tickets/5/

add status call to init script

Index: scanbd/integration/scanbd.debian
===================================================================
--- scanbd.orig/integration/scanbd.debian	2014-10-16 12:25:25.361209803 +0800
+++ scanbd/integration/scanbd.debian	2014-10-16 12:27:32.733841407 +0800
@@ -159,9 +158,11 @@
 		;;
 	esac
 	;;
+  status)
+	status_of_proc -p "$PIDFILE" "$DAEMON" "$NAME" && exit 0 || exit $?
+	;;
   *)
-	#echo "Usage: $SCRIPTNAME {start|stop|restart|reload|force-reload}" >&2
-	echo "Usage: $SCRIPTNAME {start|stop|restart|force-reload}" >&2
+	echo "Usage: $SCRIPTNAME {start|stop|restart|force-reload|status}" >&2
 	exit 3
 	;;
 esac
