Index: configure
===================================================================
diff --git a/configure b/configure
--- a/configure	(revision 213)
+++ b/configure	(revision 219)
@@ -16914,7 +16914,7 @@
 fi
 
 
-OS_CFLAGS="-Wall -Wextra -std=c99 -g"
+OS_CFLAGS="-Wall -Wextra -std=c11 -g"
 
 for ac_header in linux/limits.h \
 		 utmp.h \
Index: integration/scanbdservice.conf
===================================================================
diff --git a/integration/scanbdservice.conf b/integration/scanbdservice.conf
--- a/integration/scanbdservice.conf	(revision 213)
+++ b/integration/scanbdservice.conf	(revision 219)
@@ -3,7 +3,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/99-saned.rules
===================================================================
diff --git a/integration/99-saned.rules b/integration/99-saned.rules
--- a/integration/99-saned.rules	(revision 213)
+++ b/integration/99-saned.rules	(revision 219)
@@ -2,7 +2,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/systemd/scanbm@.service
===================================================================
diff --git a/integration/systemd/scanbm@.service b/integration/systemd/scanbm@.service
--- a/integration/systemd/scanbm@.service	(revision 213)
+++ b/integration/systemd/scanbm@.service	(revision 219)
@@ -2,7 +2,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/systemd/de.kmux.scanbd.server.service
===================================================================
diff --git a/integration/systemd/de.kmux.scanbd.server.service b/integration/systemd/de.kmux.scanbd.server.service
--- a/integration/systemd/de.kmux.scanbd.server.service	(revision 213)
+++ b/integration/systemd/de.kmux.scanbd.server.service	(revision 219)
@@ -2,7 +2,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/systemd/scanbm@.service.sane-git
===================================================================
diff --git a/integration/systemd/scanbm@.service.sane-git b/integration/systemd/scanbm@.service.sane-git
--- a/integration/systemd/scanbm@.service.sane-git	(revision 213)
+++ b/integration/systemd/scanbm@.service.sane-git	(revision 219)
@@ -2,7 +2,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/systemd/scanbd.service
===================================================================
diff --git a/integration/systemd/scanbd.service b/integration/systemd/scanbd.service
--- a/integration/systemd/scanbd.service	(revision 213)
+++ b/integration/systemd/scanbd.service	(revision 219)
@@ -2,7 +2,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/systemd/scanbm.socket
===================================================================
diff --git a/integration/systemd/scanbm.socket b/integration/systemd/scanbm.socket
--- a/integration/systemd/scanbm.socket	(revision 213)
+++ b/integration/systemd/scanbm.socket	(revision 219)
@@ -2,7 +2,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/scanbd.debian
===================================================================
diff --git a/integration/scanbd.debian b/integration/scanbd.debian
--- a/integration/scanbd.debian	(revision 213)
+++ b/integration/scanbd.debian	(revision 219)
@@ -4,7 +4,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/FreeBSD.rc
===================================================================
diff --git a/integration/FreeBSD.rc b/integration/FreeBSD.rc
--- a/integration/FreeBSD.rc	(revision 213)
+++ b/integration/FreeBSD.rc	(revision 219)
@@ -3,7 +3,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: integration/README
===================================================================
diff --git a/integration/README b/integration/README
--- a/integration/README	(revision 213)
+++ b/integration/README	(revision 219)
@@ -2,7 +2,7 @@
 
 scanbd - KMUX scanner button daemon
 
-Copyright (C) 2008, 2009, 2010, 2011, 2012, 2013  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 
 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
Index: conf/scanner.d/canon.conf
===================================================================
diff --git a/conf/scanner.d/canon.conf b/conf/scanner.d/canon.conf
--- a/conf/scanner.d/canon.conf	(revision 213)
+++ b/conf/scanner.d/canon.conf	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: conf/scanner.d/pixma.conf
===================================================================
diff --git a/conf/scanner.d/pixma.conf b/conf/scanner.d/pixma.conf
--- a/conf/scanner.d/pixma.conf	(revision 213)
+++ b/conf/scanner.d/pixma.conf	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: conf/scanner.d/snapscan.conf
===================================================================
diff --git a/conf/scanner.d/snapscan.conf b/conf/scanner.d/snapscan.conf
--- a/conf/scanner.d/snapscan.conf	(revision 213)
+++ b/conf/scanner.d/snapscan.conf	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: conf/scanner.d/hp.conf
===================================================================
diff --git a/conf/scanner.d/hp.conf b/conf/scanner.d/hp.conf
--- a/conf/scanner.d/hp.conf	(revision 213)
+++ b/conf/scanner.d/hp.conf	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: conf/scanner.d/avision.conf
===================================================================
diff --git a/conf/scanner.d/avision.conf b/conf/scanner.d/avision.conf
--- a/conf/scanner.d/avision.conf	(revision 213)
+++ b/conf/scanner.d/avision.conf	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: conf/scanner.d/fujitsu.conf
===================================================================
diff --git a/conf/scanner.d/fujitsu.conf b/conf/scanner.d/fujitsu.conf
--- a/conf/scanner.d/fujitsu.conf	(revision 213)
+++ b/conf/scanner.d/fujitsu.conf	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: conf/scanbd.conf
===================================================================
diff --git a/conf/scanbd.conf b/conf/scanbd.conf
--- a/conf/scanbd.conf	(revision 213)
+++ b/conf/scanbd.conf	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbuttond/interface/Makefile.simple
===================================================================
diff --git a/src/scanbuttond/interface/Makefile.simple b/src/scanbuttond/interface/Makefile.simple
--- a/src/scanbuttond/interface/Makefile.simple	(revision 213)
+++ b/src/scanbuttond/interface/Makefile.simple	(revision 219)
@@ -3,7 +3,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: src/scanbuttond/Makefile.simple
===================================================================
diff --git a/src/scanbuttond/Makefile.simple b/src/scanbuttond/Makefile.simple
--- a/src/scanbuttond/Makefile.simple	(revision 213)
+++ b/src/scanbuttond/Makefile.simple	(revision 219)
@@ -3,7 +3,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: src/scanbuttond/backends/Makefile.simple
===================================================================
diff --git a/src/scanbuttond/backends/Makefile.simple b/src/scanbuttond/backends/Makefile.simple
--- a/src/scanbuttond/backends/Makefile.simple	(revision 213)
+++ b/src/scanbuttond/backends/Makefile.simple	(revision 219)
@@ -3,7 +3,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: src/scanbd/scanbd.h
===================================================================
diff --git a/src/scanbd/scanbd.h b/src/scanbd/scanbd.h
--- a/src/scanbd/scanbd.h	(revision 213)
+++ b/src/scanbd/scanbd.h	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/udev.c
===================================================================
diff --git a/src/scanbd/udev.c b/src/scanbd/udev.c
--- a/src/scanbd/udev.c	(revision 213)
+++ b/src/scanbd/udev.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/slog.c
===================================================================
diff --git a/src/scanbd/slog.c b/src/scanbd/slog.c
--- a/src/scanbd/slog.c	(revision 213)
+++ b/src/scanbd/slog.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/config.c
===================================================================
diff --git a/src/scanbd/config.c b/src/scanbd/config.c
--- a/src/scanbd/config.c	(revision 213)
+++ b/src/scanbd/config.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/udev.h
===================================================================
diff --git a/src/scanbd/udev.h b/src/scanbd/udev.h
--- a/src/scanbd/udev.h	(revision 213)
+++ b/src/scanbd/udev.h	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/slog.h
===================================================================
diff --git a/src/scanbd/slog.h b/src/scanbd/slog.h
--- a/src/scanbd/slog.h	(revision 213)
+++ b/src/scanbd/slog.h	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/config.h
===================================================================
diff --git a/src/scanbd/config.h b/src/scanbd/config.h
--- a/src/scanbd/config.h	(revision 213)
+++ b/src/scanbd/config.h	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/scanbuttond_loader.c
===================================================================
diff --git a/src/scanbd/scanbuttond_loader.c b/src/scanbd/scanbuttond_loader.c
--- a/src/scanbd/scanbuttond_loader.c	(revision 213)
+++ b/src/scanbd/scanbuttond_loader.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/daemonize.c
===================================================================
diff --git a/src/scanbd/daemonize.c b/src/scanbd/daemonize.c
--- a/src/scanbd/daemonize.c	(revision 213)
+++ b/src/scanbd/daemonize.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/scanbd_dbus.h
===================================================================
diff --git a/src/scanbd/scanbd_dbus.h b/src/scanbd/scanbd_dbus.h
--- a/src/scanbd/scanbd_dbus.h	(revision 213)
+++ b/src/scanbd/scanbd_dbus.h	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/scanbuttond_loader.h
===================================================================
diff --git a/src/scanbd/scanbuttond_loader.h b/src/scanbd/scanbuttond_loader.h
--- a/src/scanbd/scanbuttond_loader.h	(revision 213)
+++ b/src/scanbd/scanbuttond_loader.h	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/testscanbuttond.c
===================================================================
diff --git a/src/scanbd/testscanbuttond.c b/src/scanbd/testscanbuttond.c
--- a/src/scanbd/testscanbuttond.c	(revision 213)
+++ b/src/scanbd/testscanbuttond.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/Makefile.simple
===================================================================
diff --git a/src/scanbd/Makefile.simple b/src/scanbd/Makefile.simple
--- a/src/scanbd/Makefile.simple	(revision 213)
+++ b/src/scanbd/Makefile.simple	(revision 219)
@@ -3,7 +3,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: src/scanbd/scanbuttond_wrapper.c
===================================================================
diff --git a/src/scanbd/scanbuttond_wrapper.c b/src/scanbd/scanbuttond_wrapper.c
--- a/src/scanbd/scanbuttond_wrapper.c	(revision 213)
+++ b/src/scanbd/scanbuttond_wrapper.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/sane.c
===================================================================
diff --git a/src/scanbd/sane.c b/src/scanbd/sane.c
--- a/src/scanbd/sane.c	(revision 213)
+++ b/src/scanbd/sane.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -194,7 +194,12 @@
     // handle h
     // if option can't be found or other catastrophy happens, the
     // value 0 gets returned
+#if ((__STDC_VERSION__  - 0) < 201112L) || (((__GNUC__ - 0) < 5) && ((__GNUC_MINOR__ - 0) < 9))
+    sane_opt_value_t res;
+#else
     sane_opt_value_t res = {};
+#endif
+
     sane_option_value_init(&res);
 
     const SANE_Option_Descriptor* odesc = NULL;
Index: src/scanbd/scanbuttond_wrapper.h
===================================================================
diff --git a/src/scanbd/scanbuttond_wrapper.h b/src/scanbd/scanbuttond_wrapper.h
--- a/src/scanbd/scanbuttond_wrapper.h	(revision 213)
+++ b/src/scanbd/scanbuttond_wrapper.h	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/scanbd.c
===================================================================
diff --git a/src/scanbd/scanbd.c b/src/scanbd/scanbd.c
--- a/src/scanbd/scanbd.c	(revision 213)
+++ b/src/scanbd/scanbd.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: src/scanbd/dbus.c
===================================================================
diff --git a/src/scanbd/dbus.c b/src/scanbd/dbus.c
--- a/src/scanbd/dbus.c	(revision 213)
+++ b/src/scanbd/dbus.c	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -61,8 +61,13 @@
     }
 
     if (argv != NULL) {
+#if ((__STDC_VERSION__  - 0) < 201112L) || (((__GNUC__ - 0) < 5) && ((__GNUC_MINOR__ - 0) < 9))
+        DBusMessageIter args;
+        DBusMessageIter sub;
+#else
         DBusMessageIter args = {};
         DBusMessageIter sub = {};
+#endif
         dbus_message_iter_init_append(signal, &args);
         if (dbus_message_iter_open_container(&args, DBUS_TYPE_ARRAY,
                                              DBUS_TYPE_STRING_AS_STRING,
@@ -118,7 +123,11 @@
     }
 
     if (arg != NULL) {
+#if ((__STDC_VERSION__  - 0) < 201112L) || (((__GNUC__ - 0) < 5) && ((__GNUC_MINOR__ - 0) < 9))
+        DBusMessageIter args;
+#else
         DBusMessageIter args = {};
+#endif
         dbus_message_iter_init_append(signal, &args);
         slog(SLOG_DEBUG, "append string %s to signal %s", arg, signal_name);
         if (dbus_message_iter_append_basic(&args, DBUS_TYPE_STRING, &arg) != TRUE) {
@@ -136,14 +145,25 @@
 }
 
 static void hook_device_ex(const char *param, const char *action_name, const char *dev_name) {
+    slog(SLOG_DEBUG, "hook_device_ex");
+    assert(dev_name);
+    assert(param);
+    assert(action_name);
+
+    slog(SLOG_DEBUG, "hook_device_ex: p: %s");
+    slog(SLOG_DEBUG, "hook_device_ex: a: %s");
+    slog(SLOG_DEBUG, "hook_device_ex: d: %s");
+
     cfg_t* cfg_sec_global = cfg_getsec(cfg, C_GLOBAL);
     assert(cfg_sec_global);
     const char* script = cfg_getstr(cfg_sec_global, param);
 
     if (!script || (strlen(script) == 0)) {
+        slog(SLOG_INFO, "No hook script for inserting device: %s", dev_name);
         //script = SCANBD_NULL_STRING;
         return; // No hook script, nothing for us to do here.
-    }       
+    }
+    slog(SLOG_INFO, "Using hook script %s for inserting device: %s", script, dev_name);
 
     cfg_t* global_envs = cfg_getsec(cfg_sec_global, C_ENVIRONMENT);
     // number of env-vars =
@@ -233,6 +253,8 @@
 
     char *script_abs = make_script_path_abs(script);
     assert(script_abs);
+    slog(SLOG_DEBUG, "Hook script path: %s for inserting device: %s", script_abs, dev_name);
+
     if (strcmp(script_abs, SCANBD_NULL_STRING) != 0) {
         pid_t cpid = -1;
         if ((cpid = fork()) < 0) {
@@ -267,6 +289,7 @@
 }
 
 static void hook_device_insert(const char *dev_name) {
+    slog(SLOG_DEBUG, "hook_device_insert");
     hook_device_ex(C_DEVICE_INSERT_SCRIPT, "insert", dev_name);
 }
 
@@ -523,11 +546,15 @@
 }
 
 static void dbus_method_trigger(DBusMessage *message) {
-    DBusMessageIter args = {};
+#if ((__STDC_VERSION__  - 0) < 201112L) || (((__GNUC__ - 0) < 5) && ((__GNUC_MINOR__ - 0) < 9))
+        DBusMessageIter args;
+#else
+        DBusMessageIter args = {};
+#endif
 
     dbus_uint32_t device = -1;
     dbus_uint32_t action = -1;
-    
+
     if (!dbus_message_iter_init(message, &args)) {
         slog(SLOG_WARN, "trigger has no arguments");
         return;
@@ -565,7 +592,7 @@
                                       void *user_data) {
     (void)connection;
     (void)user_data;
-    
+
     slog(SLOG_DEBUG, "message_func");
     DBusMessage* reply = NULL;
     if (dbus_message_is_method_call(message,
@@ -623,7 +650,7 @@
     sigset_t mask;
     sigfillset(&mask);
     pthread_sigmask(SIG_BLOCK, &mask, NULL);
-    
+
     cfg_t* cfg_sec_global = NULL;
     cfg_sec_global = cfg_getsec(cfg, C_GLOBAL);
     assert(cfg_sec_global);
@@ -664,7 +691,7 @@
     if (!dbus_threads_init_default()) {
         slog(SLOG_ERROR, "DBus thread initialization failure");
     }
-    
+
     DBusError dbus_error;
     dbus_error_init(&dbus_error);
 
@@ -697,7 +724,7 @@
     }
 
     libhal_ctx_set_dbus_connection(hal_ctx, conn);
-    
+
     libhal_ctx_set_device_added(hal_ctx, hal_device_added);
     libhal_ctx_set_device_removed(hal_ctx, hal_device_removed);
 
@@ -733,7 +760,7 @@
         }
     }
     assert(conn);
-    
+
     if (!conn) {
         slog(SLOG_DEBUG, "No dbus connection");
         return;
@@ -740,7 +767,7 @@
     }
 
     DBusError dbus_error;
-    
+
     dbus_error_init(&dbus_error);
     if (dbus_connection_register_object_path(conn, SCANBD_DBUS_OBJECTPATH,
                                              &dbus_vtable, NULL) == FALSE) {
@@ -816,7 +843,11 @@
     assert(msg);
 
     if (value != NULL) {
+#if ((__STDC_VERSION__  - 0) < 201112L) || (((__GNUC__ - 0) < 5) && ((__GNUC_MINOR__ - 0) < 9))
+        DBusMessageIter args;
+#else
         DBusMessageIter args = {};
+#endif
         dbus_message_iter_init_append(msg, &args);
         if (dbus_message_iter_append_basic(&args, DBUS_TYPE_STRING, value) != TRUE) {
             slog(SLOG_ERROR, "Can't compose message");
@@ -823,7 +854,7 @@
             return;
         }
     }
-    
+
     DBusPendingCall* pending = NULL;
 
     // send message and get a handle for a reply
@@ -852,11 +883,15 @@
     }
     dbus_pending_call_unref(pending);
 
-    DBusMessageIter args = {};
+#if ((__STDC_VERSION__  - 0) < 201112L) || (((__GNUC__ - 0) < 5) && ((__GNUC_MINOR__ - 0) < 9))
+        DBusMessageIter args;
+#else
+        DBusMessageIter args = {};
+#endif
     if (!dbus_message_iter_init(reply, &args)) {
         slog(SLOG_INFO, "Reply has no arguments");
     }
-    
+
     return;
 }
 
@@ -886,7 +921,11 @@
 
     dbus_uint32_t dev = device;
     dbus_uint32_t act = action;
-    DBusMessageIter args = {};
+#if ((__STDC_VERSION__  - 0) < 201112L) || (((__GNUC__ - 0) < 5) && ((__GNUC_MINOR__ - 0) < 9))
+        DBusMessageIter args;
+#else
+        DBusMessageIter args = {};
+#endif
     dbus_message_iter_init_append(msg, &args);
     if (dbus_message_iter_append_basic(&args, DBUS_TYPE_UINT32, &dev) != TRUE) {
         slog(SLOG_ERROR, "Can't compose message");
@@ -896,7 +935,7 @@
         slog(SLOG_ERROR, "Can't compose message");
         return;
     }
-    
+
     DBusPendingCall* pending = NULL;
 
     // send message and get a handle for a reply
@@ -928,6 +967,6 @@
     if (!dbus_message_iter_init(reply, &args)) {
         slog(SLOG_INFO, "Reply has no arguments");
     }
-    
+
     return;
 }
Index: src/scanbd/common.h
===================================================================
diff --git a/src/scanbd/common.h b/src/scanbd/common.h
--- a/src/scanbd/common.h	(revision 213)
+++ b/src/scanbd/common.h	(revision 219)
@@ -3,7 +3,7 @@
  *
  *  scanbd - KMUX scanner button daemon
  *
- *  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+ *  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
  *
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
Index: scripts/scanadf.script
===================================================================
diff --git a/scripts/scanadf.script b/scripts/scanadf.script
--- a/scripts/scanadf.script	(revision 213)
+++ b/scripts/scanadf.script	(revision 219)
@@ -4,7 +4,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: scripts/test.script
===================================================================
diff --git a/scripts/test.script b/scripts/test.script
--- a/scripts/test.script	(revision 213)
+++ b/scripts/test.script	(revision 219)
@@ -4,7 +4,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: scripts/copy.script
===================================================================
diff --git a/scripts/copy.script b/scripts/copy.script
--- a/scripts/copy.script	(revision 213)
+++ b/scripts/copy.script	(revision 219)
@@ -4,7 +4,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: scripts/example.script
===================================================================
diff --git a/scripts/example.script b/scripts/example.script
--- a/scripts/example.script	(revision 213)
+++ b/scripts/example.script	(revision 219)
@@ -4,7 +4,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: Makefile.include
===================================================================
diff --git a/Makefile.include b/Makefile.include
--- a/Makefile.include	(revision 213)
+++ b/Makefile.include	(revision 219)
@@ -3,7 +3,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2012  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
@@ -197,7 +197,7 @@
 # Define common options
 # ======================
 #
-CFLAGS   += -Wall -Wextra -std=c99 -g
+CFLAGS   += -Wall -Wextra -std=c11 -g
 LDLIBS  += -lconfuse -lpthread -ldbus-1
 CPPFLAGS += -DSCANBD_CFG_DIR=\"$(SCANBD_CFG_DIR)\" -DSCANBUTTOND_LIB_DIR=\"$(SCANBUTTOND_LIB_DIR)\" \
 	-DSANED_PATH=\"$(SANED_PATH)\"
Index: Makefile.conf
===================================================================
diff --git a/Makefile.conf b/Makefile.conf
--- a/Makefile.conf	(revision 213)
+++ b/Makefile.conf	(revision 219)
@@ -2,7 +2,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2012  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: Makefile.simple
===================================================================
diff --git a/Makefile.simple b/Makefile.simple
--- a/Makefile.simple	(revision 213)
+++ b/Makefile.simple	(revision 219)
@@ -2,7 +2,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: doc/Makefile.simple
===================================================================
diff --git a/doc/Makefile.simple b/doc/Makefile.simple
--- a/doc/Makefile.simple	(revision 213)
+++ b/doc/Makefile.simple	(revision 219)
@@ -3,7 +3,7 @@
 #
 #  scanbd - KMUX scanner button daemon
 #
-#  Copyright (C) 2008 - 2015  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+#  Copyright (C) 2008 - 2016 Wilhelm Meier (wilhelm.meier@fh-kl.de)
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
Index: doc/README.txt
===================================================================
diff --git a/doc/README.txt b/doc/README.txt
--- a/doc/README.txt	(revision 213)
+++ b/doc/README.txt	(revision 219)
@@ -2,7 +2,7 @@
 
 scanbd - KMUX scanner button daemon
 
-Copyright (C) 2008, 2009, 2010, 2011, 2012, 2013  Wilhelm Meier (wilhelm.meier@fh-kl.de)
+Copyright (C) 2008 - 2016  Wilhelm Meier (wilhelm.meier@hs-kl.de)
 
 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
